src/VecSim/vec_sim_interface.h

```diff
@@ -220,6 +220,11 @@ struct VecSimIndexInterface : public VecsimBaseObject {
      */
     virtual inline void setLastSearchMode(VecSearchMode mode) = 0;
 
+    /**
+     * @brief Run async garbage collection for tiered async index.
+     */
+    virtual void runGC() = 0;
```

Comment:
Consider implementing the no-op on the abstract class so we will need to have no-op on flat/hnsw etc..